src/main/resources/swagger/api-docs.yaml

```diff
@@ -795,16 +821,16 @@ paths:
         - $ref: '#/parameters/workspaceNamespaceParam'
         - $ref: '#/parameters/workspaceNameParam'
         - name: discoverableGroups
-          description: List of group names, or empty list for no restrictions
+          description: Json array of group names, or empty array for no restrictions
           required: true
           in: body
+          schema:
+            $ref: '#/definitions/StringArray'
       responses:
         200:
-          description: The groups that have been assigned to be able to discover the dataset
+          description: The workspace, after assigning groups to be able to discover the dataset
           schema:
-            type: array
-            items:
-              type: string
+            $ref: '#/definitions/Workspace'
```

Comment:
Good catch. You'll also get a 400 if you input something that isn't a `List[String]`. I added the 400 response code.